nixos/doc/manual/development/option-def.section.md

```diff
@@ -123,3 +123,29 @@ they were declared in separate modules. This can be done using
     ];
 }
 ```
+
+## Defining Over All Elements/Attributes {#sec-option-definitions-for-all-values}
+
+For options that take lists or attrsets, it may be useful to define defaults or
+overrides on all of the elements or attributes of that option independently of
+the population of the list or set. For example, one module can define a list of
+`swapDevices`, while another module can define that for every element of the list
+of `swapDevices`, `swapDevices.*.options` list should have a particular value. This
+can be done using `mkForAllValues`:
+
+```nix
+{
+  swapDevices = mkForAllValues {
+    options = mkDefault [ /* ... */ ];
```

Comment:
I'd be worried about the home directory example because, as implemented, it uses the value of one attribute in a `mkIf` used to compute another. This is doable with `mkForAllValues` as well but there is some subtlety to it and I don't want that to distract.

Is the issue with `swapDevices.*.options` just the word `options`? Can we take `swapDevices.*.discardPolicy` instead, or something? Or how about `services.wordpress.sites.<name>.uploadsDir`, if we want to showcase the parameter? (I'm leaning toward both.)